### PR TITLE
indexing.rst edits

### PR DESCRIPTION
**Pull request by shoyer**

### CODE REVIEW

File: doc/indexing.rst [hunk 1]
```diff
@@ -648,28 +656,43 @@ dimensions or use the ellipsis in the ``loc`` specifier, e.g. in the example
...
 
-5. If an indexer-DataArray has coordinates, attached them to the indexed object.
+5. If any indexer DataArray has coordinates and no coordinate with the
+   same name exists, attach them to the indexed object.
 
 .. note::
 
-  + There should not be a conflict between the coordinates of indexer- and indexed- DataArrays. In v.0.10.0, xarray raises ``FutureWarning`` if there is such a conflict, but in the next major release, it will raise an Error.
+  - ``IndexError`` is raised if there is a conflict between dimension
+     coordinates of indexer- and indexed- DataArrays.
```
Comment (fujiisoup) [inline doc/indexing.rst]: Should we raise `IndexError` for coordinate conflict?
The current code just neglects the conflicted coordinates.

**Review by fujiisoup — COMMENTED** [review]
Thank you for improving the doc. This is much smoother and more natural :)

File: doc/indexing.rst (hunk 1 above)
Comment (shoyer) [reply doc/indexing.rst]: I think we do want to raise an error here, but as you note this is not the case currently. Let's discuss this over in the main issue (where I also just raised this concern)

Comment (shoyer) [reply doc/indexing.rst]: I removed this line for now. Let's add it back in later when we settle on how it should work.

**Pull request merged**